platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r813/ProblemProgressEventCrossVersionTest.groovy

```diff
@@ -134,6 +139,76 @@ class ProblemProgressEventCrossVersionTest extends ToolingApiSpecification {
         ''                         | null            | ''                                          | null
     }
 
+    @TargetGradleVersion(">=8.13")
+    def "can serialize arbitrary additional data"() {
```

Comment:
❌ I think we need more test coverage than this. Please think of the possible scenarios:
- data coming from different sources (plugins, build scripts, workers)
- serialization fails (I expect we ignore the additional data or replace it with a warning data showing that serialization fails)
- Client requests wrong data type 
  - null
  - Object
  - WrongType
- etc

Reply:
We definitely need to make the worker usecase, but not necessarily in the same PR.